crates/bevy_ui/src/focus.rs

```diff
@@ -176,6 +177,7 @@ pub fn ui_focus_system(
             if let Some(mut interaction) = node.interaction {
                 if *interaction == Interaction::Pressed {
                     *interaction = Interaction::None;
+                    click_events.send(Click(node.entity));
```

Comment:
After some thinking I come to a conclusion that it's not equivalent. We need to register click only if mouse stays at the bottom (i.e. the state is equal to `Hovered`). What if we check if `mouse_released` on line 267 instead and emit the event?

Reply:
For whatever reason, doing:

```rust
if contains_cursor {
    Some(*entity)
} else {
    if let Some(mut interaction) = node.interaction {
        if *interaction == Interaction::Hovered || (cursor_position.is_none()) {
            interaction.set_if_neq(Interaction::None);
        }
    }
    if mouse_released {
        click_events.send(Click(node.entity));
    }
    None
}
```

Causes programs to silently crash, i might be missing something. When emitting events at line 172 it seems to work and behave as it did before.

Reply:
> For whatever reason, doing:

Sorry, I mean under `if` statement, not `else`. If understand correctly, we should do it only if the item continue to be hovered. Not sure if it's the correct place either, something to try.

>  When emitting events at line 172 it seems to work and behave as it did before.

Unfortunately, not :(
Try to click and hold RMB under a button, then move cursor outside of the button and release the RMB. The button will be considered clicked, while it should not.

Reply:
Doing

```rust
if contains_cursor {
    if mouse_released {
        click_events.send(Click(*entity));
    }
    Some(*entity)
} else {
    if let Some(mut interaction) = node.interaction {
        if *interaction == Interaction::Hovered || (cursor_position.is_none()) {
            interaction.set_if_neq(Interaction::None);
        }
    }
    None
}
```

Appears to work with the only quirk I've seen so far being that in the `game_menu` example. If i hold left click on the menu buttons, then drag my mouse away from the button and release, the buttons don't trigger. Is this the desired behaviour?

Reply:
Yes, this is exactly how it should work. You can try the same with any button in your system, it works this way :)